projects/clas12/targets/geometry.py

```diff
@@ -504,13 +461,1169 @@ def build_geometry_pol_targ(configuration):
+		gvolume.color =  "000000"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Al"
+		return gvolume
+
+	for builder in [
+		build_bonus_root_volume,
+		build_bonus_target_gas_volume,
+		build_bonus_target_wall,
+		build_bonus_target_downstream_aluminum_end_cap_r_ing,
+		build_bonus_target_downstream_end_cap_r_ing,
+	]:
+		volume = builder()
+		volume.publish(configuration)
+
+
+def build_geometry_pb_test(configuration):
+
+	z_shift = 40.85
+	cell_half_length = 293.26 / 2
+
+	def build_helium_bag():
+		# mother is a helium bag 293.26 mm long. Its center is 40.85 mm
+		r_in = 0.0
+		r_out = 10.0
+		z_center = cell_half_length - z_shift
+		gvolume = GVolume("targetCell")
+		gvolume.mother = "root"
+		gvolume.description ="Helium cell"
+		gvolume.color =  "5511111"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, cell_half_length)
+		gvolume.material = "G4_He"
+		return gvolume
+
+	def build_pb_125_microns():
+		# PB 125 microns
+		r_in = 0.0
+		r_out = 5.0
+		z_center = z_shift - cell_half_length
+		half_length = 0.0625 # (0.125 mm thick)
+		gvolume = GVolume("testPbTarget")
+		gvolume.mother = "targetCell"
+		gvolume.description ="Pb target"
+		gvolume.color =  "004488"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Pb"
+		return gvolume
+
+	def build_upstream_foil_al():
+		# Upstream Foil 50 microns Aluminum at 24.8 cm
+		z_center = 0.1 - cell_half_length   # upstream end
+		r_in = 0.0
+		r_out = 5.0 
+		half_length = 0.015 # (0.030 mm thick)
+		gvolume = GVolume("AlTargetFoilUpstream")
+		gvolume.mother = "targetCell"
+		gvolume.description ="Aluminum Upstream Foil"
+		gvolume.color =  "aaaaaa"
+		gvolume.color =  "004488"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Al"
+		return gvolume
+
+	def build_downstream_foil_al():
+		# Downstream Foil 50 microns Aluminum at 24.8 cm
+		z_center = cell_half_length - 0.1   # Downstream end
+		r_in = 0.0
+		r_out = 5.0
+		half_length = 0.015 # (0.030 mm thick)
+		gvolume = GVolume("AlTargetFoilDownstream")
+		gvolume.mother = "targetCell"
+		gvolume.description ="Aluminum Downstream Foil"
+		gvolume.color =  "aaaaaa"
+		gvolume.material = "G4_Al"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		return gvolume
+
+	def build_flux_detector():
+		# flux detector downstream of the scatter_ing chamber
+		z_center = 300.0   
+		r_out = 45.0
+		r_in = 0.0
+		half_length = 0.015 # (0.030 mm thick)
+		gvolume = GVolume("testFlux")
+		gvolume.mother = "root"
+		gvolume.description = "Flux detector"
+		gvolume.color = "009900"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_AIR"
+		# @mariakzurek: sensitivity, hit_type, identifiers are not part of the current gemc3 API
```

Comment:
@maureeungaro Please check this comment

Reply:
Actually they all are implemented, see for example the plugins / calorimeter example. None of those apply to the target though.